mbf_msgs/srv/FindValidPose.srv

```diff
@@ -0,0 +1,27 @@
+# Find the closest valid pose to the given one, within the given linear and angular tolerances.
+# A pose is considered valid if the robot footprint there is entirely inside the map and not in
+# collision nor unknown space.
```

Comment:
I think this default definition of `vaild pose` makes sense, but I guess technically, we could let the user decide what they consider a valid pose... Maybe allowing poses partially outside the map is an unneeded edge case; I don't know how commonly unknown space is used. Anyway, could just be a future feature to add a flag to allow poses where the footprint contains cells with unknown cost

Reply:
Maybe the conclusion from all my 3 other comments could be:
- remove the boolean `valid` field in the result
- return `FREE` if a pose completely in free space (including inscribed) is found, else return one partially in UNKNOWN, else return one partially outside the map, else return collision. The user can then decide themselves whether they want to use the pose or not based on the return code?

Reply:
yes,,, makes sense;  or much simpler
```
uint8                      FREE      =  0    # found pose is completely in traversable space
uint8                      INSCRIBED =  1    # found pose is partially in inscribed space
uint8                      INVALID =  2    # no valid pose found
```
and remove valid flag